packages/gestalt/src/SelectList.js

```diff
@@ -115,77 +92,63 @@ export default class SelectList extends React.Component<Props, State> {
-        <select
-          aria-describedby={
-            errorMessage && this.state.focused ? `${id}-gestalt-error` : null
-          }
-          aria-invalid={errorMessage ? 'true' : 'false'}
-          className={classes}
-          disabled={disabled}
-          id={id}
-          name={name}
-          onBlur={this.handleBlur}
-          onFocus={this.handleFocus}
-          onChange={this.handleOnChange}
-          ref={c => {
-            this.select = c;
-          }}
-          value={value}
+          position="relative"
+          width="100%"
         >
-          {placeholder &&
-            !value && (
-              <option selected disabled value hidden>
-                {placeholder}
+          <Box
+            alignItems="center"
+            bottom
+            dangerouslySetInlineStyle={{
+              __style: { paddingRight: 14, paddingTop: 2 },
+            }}
+            display="flex"
+            position="absolute"
+            right
+            top
+          >
+            <Icon
+              icon="arrow-down"
+              size={12}
+              color={disabled ? 'gray' : 'darkGray'}
+              accessibilityLabel=""
+            />
+          </Box>
+          <select
+            aria-describedby={
+              errorMessage && this.state.focused ? formatErrorId(id) : null
+            }
+            aria-invalid={errorMessage ? 'true' : 'false'}
+            className={classes}
+            disabled={disabled}
+            id={id}
+            name={name}
+            onBlur={this.handleOnChange}
+            onChange={this.handleOnChange}
+            ref={c => {
+              this.select = c;
+            }}
+            value={value}
+          >
+            {placeholder &&
+              !value && (
+                <option selected disabled value hidden>
+                  {placeholder}
+                </option>
+              )}
+            {options.map(option => (
+              <option key={option.value} value={option.value}>
+                {option.label}
               </option>
-            )}
-          {options.map(option => (
-            <option key={option.value} value={option.value}>
-              {option.label}
-            </option>
-          ))}
-        </select>
-        {errorMessage &&
-          this.state.errorIsOpen && (
-            <Flyout
-              anchor={this.select}
-              color="orange"
-              idealDirection={idealErrorDirection}
-              onDismiss={() => this.setState({ errorIsOpen: false })}
-              shouldFocus={false}
-              size="sm"
-            >
-              <Box padding={3}>
-                <Text bold color="white">
-                  <span id={`${id}-gestalt-error`}>{errorMessage}</span>
-                </Text>
-              </Box>
-            </Flyout>
-          )}
+            ))}
+          </select>
+        </Box>
+
+        {errorMessage && <ErrorMessage errorMessage={errorMessage} id={id} />}
```

Comment:
This seems a little funny to me. `<ErrorMessage>` isn't composable because it has the `marginTop` (that assumes the usage next to the calling component. If we removed that out to the SelectList then ErrorMessage really just becomes text colored orange. The value of the abstraction breaks down a bit then.

An alternative design is to have ErrorMessage take a prop which is the element to wrap:

```jsx
<ErrorMessage field={<SelectList /> message="asdf" />
```

I like this because a.) it removes an unnessecary `<Box>` on elements when they're not erroring b.) consumers can choose _how_ to format their error messages without them always being below. This approach is similar to experimental "Field" approaches we've seen around the place and helps clarify the naming (`ErrorMessage` is vague yet the component is specific).

Reply:
Does changing to a more specific `FormErrorMessage` component resolve this? If the component is indicating that's it's pretty specifically to be used as an error message for forms, that makes the `marginTop` seem less awkward…and this is intended to be a purely "internal" component, not one exposed to users. I get the flexibility of an error component that wraps its `children` and can be arbitrarily styled, but that seems beyond the scope of this change.

Reply:
IMHO one of the benefits of the previous implementation was that the error message didn't change the field's layout. It was always the same. Now, if you want to color the field yellow, the layout will change and a message is *required*. A good example of where this mightn't work is in top banners where it might be more convenient to display an error message side by side with a field. To reduce the scope of this change the "errorMessage" part of the component could be deprecated and the yellow text added to experimental fields. Another stage would to be to take those fields and make them less experimental :)

Reply:
Good point, and probably something we should / may need to address in the future. For the sake of expediency, I'm going to modify the existing paradigm rather than introduce a new one.

You're definitely right about moving the Box with marginTop out of FormErrorMessage.